overlord/snapstate/snapstate.go

```diff
@@ -131,8 +131,21 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 
 		if experimentalRefreshAppAwareness {
 			if err := SoftNothingRunningRefreshCheck(info); err != nil {
-				// TODO Remember the inhibition time in snap state.
-				return nil, err
```

Comment:
we probably want a comment here that explains  that because this might manipulate the snap state it cannot be earlier than after the conflict check